frame/tidefi-stake/src/lib.rs

```diff
@@ -444,8 +444,9 @@ pub mod pallet {
 
       // 3. Check the expiration and if we are forcing it (queue)
       let expected_block_expiration = stake.initial_block + stake.duration;
```

Comment:
As we are here, can you do something like this instead;

```diff
-    let expected_block_expiration = stake.initial_block + stake.duration;
+   let expected_block_expiration = stake.initial_block.saturating_add(stake.duration);
```

It'll prevent an overflow and a panic if the extrinsic is created with crazy values

Thanks !

Reply:
Safe maths is applied.